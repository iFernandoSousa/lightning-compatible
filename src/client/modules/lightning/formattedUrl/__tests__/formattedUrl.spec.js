/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import Element from 'c/formattedUrl';
import { shadowQuerySelector } from 'lightning/testUtils';

const createComponent = (params = {}) => {
    const element = createElement('c-formatted-url', { is: Element });
    Object.assign(element, params);
    document.body.appendChild(element);
    return element;
};

describe('c-formatted-url', () => {
    it('default', () => {
        const element = createComponent();
        expect(element).toMatchSnapshot();
    });

    it('url without prefix', () => {
        const element = createComponent({
            value: 'foo/sss',
            target: '_blank'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('url has http prefix', () => {
        const element = createComponent({
            value: 'http://foo/nns',
            target: '_blank'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('url has https prefix', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: '_blank'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('url has ftp prefix', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: '_blank',
            title: '123'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('url has title', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: '_blank',
            tooltip: '123'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('url has title, and target is _self', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: '_self',
            title: '123',
            label: 'TheLabel'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('url has title, and target is a named iframe', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: 'someIFrame',
            tooltip: '123',
            label: 'TheLabel'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('label changes', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: 'someIFrame',
            tooltip: '123'
        });

        return Promise.resolve()
            .then(() => {
                expect(element).toMatchSnapshot();
                element.label = 'A label';
            })
            .then(() => {
                expect(element).toMatchSnapshot();
                element.label = '';
            })
            .then(() => {
                expect(element).toMatchSnapshot();
            });
    });

    it('tooltip changes', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: 'someIFrame',
            tooltip: '123'
        });

        return Promise.resolve()
            .then(() => {
                expect(element).toMatchSnapshot();
                element.tooltip = '';
            })
            .then(() => {
                expect(element).toMatchSnapshot();
                element.label = '123';
            })
            .then(() => {
                expect(element).toMatchSnapshot();
            });
    });

    it('target changes', () => {
        const element = createComponent({
            value: 'https://foo/nns',
            target: 'someIFrame'
        });

        return Promise.resolve()
            .then(() => {
                expect(element).toMatchSnapshot();
                element.target = '';
            })
            .then(() => {
                expect(element).toMatchSnapshot();
                element.target = '_blank';
            })
            .then(() => {
                expect(element).toMatchSnapshot();
            });
    });

    it('sets tabIndex', () => {
        const element = createComponent({
            value: 'somelink.com',
            tabIndex: '0'
        });

        return Promise.resolve().then(() => {
            expect(element.getAttribute('tabindex')).toBeFalsy();
            expect(
                shadowQuerySelector(element, 'a').getAttribute('tabindex')
            ).toBe('0');
        });
    });
});
