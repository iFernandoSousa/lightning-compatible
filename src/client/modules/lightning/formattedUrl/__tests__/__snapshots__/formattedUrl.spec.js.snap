// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-formatted-url default 1`] = `
<c-formatted-url>
  #shadow-root(open)
</c-formatted-url>
`;

exports[`c-formatted-url label changes 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
      title="123"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url label changes 2`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
      title="123"
    >
      A label
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url label changes 3`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
      title="123"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url target changes 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url target changes 2`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target=""
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url target changes 3`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="_blank"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url tooltip changes 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
      title="123"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url tooltip changes 2`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
      title=""
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url tooltip changes 3`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
      title=""
    >
      123
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url url has ftp prefix 1`] = `
<c-formatted-url
  title="123"
>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="_blank"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url url has http prefix 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="http://foo/nns"
      target="_blank"
    >
      http://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url url has https prefix 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="_blank"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url url has title 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="_blank"
      title="123"
    >
      https://foo/nns
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url url has title, and target is _self 1`] = `
<c-formatted-url
  title="123"
>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="_self"
    >
      TheLabel
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url url has title, and target is a named iframe 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="https://foo/nns"
      target="someIFrame"
      title="123"
    >
      TheLabel
    </a>
</c-formatted-url>
`;

exports[`c-formatted-url url without prefix 1`] = `
<c-formatted-url>
  #shadow-root(open)
    <a
      href="http://foo/sss"
      target="_blank"
    >
      http://foo/sss
    </a>
</c-formatted-url>
`;
